### DISCUSSION TITLE
Can I use lockFileMaintenance only for one out of many supported lockfiles in a repository?

### DISCUSSION THREAD

**Discussion by jelhan** [renovatebot/renovate#25219] [Request Help]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

I'm using RenovateBot on a repository, which contains a `yarn.lock` and a `composer.lock`. I want to configure RenovateBot to use `lockFileMaintenance` for the `yarn.lock` file but not for the `composer.lock`. Is this possible?

### What you tried so far.

This is my current configuration:

```json
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "config:base",
    ":maintainLockFilesWeekly",
    ":preserveSemverRanges"
  ],
  "ignorePaths": [
    "lib/**"
  ],
  "postUpdateOptions": ["yarnDedupeHighest"]
}
```

I had a look at [configuration options for `lockFileMaintenance`](https://docs.renovatebot.com/configuration-options/#lockfilemaintenance) but I haven't noticed anything, which allows me to configure the lock file for which it should run. It seems to always run for all supported lock files within the repository.

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by rarkins** [comment]
You should be able to achieve this using a packageRules entries with matchManagers or matchFileNames

**Reply by jelhan** [reply]
I tried this out. Either `lockfileMaintenance` is not supported in `packageRules` or I'm too stupid to configure it correctly.

This is my configuration:

```json
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "config:base",
    ":preserveSemverRanges"
  ],
  "packageRules": [
    {
      "matchManagers": ["npm"],
      "lockFileMaintenance": {
        "enabled": true,
        "extends": [
          "schedule:weekly"
        ]
      }
    }
  ]
}
```

I expect to see an entry for lockfile maintenance in the dependency dashboard. But I don't see it. It seems to be ignored.

I have setup a simplified reproduction here: https://github.com/jelhan/renovate-bot-lockfile-maintenance-with-package-rules/issues/1

Edit: Noticed that my RenovateBot configuration was wrong indeed. Fixed it.

**Reply by rarkins** [reply]
Please remove all `extends` from within `packageRules` to further simplify it and rule those out as a possible cause

**Reply by jelhan** [reply]
I simplified the configuration as requested:

```json
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "config:base"
  ],
  "packageRules": [
    {
      "matchManagers": ["npm"],
      "lockFileMaintenance": {
        "enabled": true,
        "schedule": ["before 4am on monday"],
      }
    }
  ]
}
```

`lockFileMaintenance` seems to be still ignored.

**Reply by rarkins** [reply]
It has a config error on this line, so I don't think it's truly tested: https://github.com/jelhan/renovate-bot-lockfile-maintenance-with-package-rules/blob/501ff729d6233510c7ee6864676b6765d885369c/renovate.json#L11

**Reply by jelhan** [reply]
Good catch. I fixed that typo but still doesn't see a PR for lockfile maintenance in the dependency dashboard issue.

**Reply by rarkins** [reply]
I think I found a way, see https://github.com/renovate-reproductions/25219/issues/2 and the config change I made there. Renovate reports that the lock file doesn't need updating, which is maybe correct, but at least now it's trying
